### PR TITLE
Stop clearing cache when incompatible plugins list is updated

### LINKED ISSUE #6077: The whole cache is cleared when the incompatible plugins dynamic list is the only one updated
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version :heavy_check_mark: 
- Used the search feature to ensure that the bug hasn’t been reported before :heavy_check_mark: 

**Describe the bug**

When:
1. Clicking the "UPDATE LISTS" button in the tools tab and 
2. the incompatible plugins list is the only one that has been updated 

the whole cache is cleared.

This isn't necessary since the content of the specific dynamic list doesn't affect the front end.

**To Reproduce**

1. Add or change an entry in the incompatible plugins list in the staging backend.
2. Make sure URLs are cached in the `/cache/wp-rocket/` folder.
3. Click on the "UPDATE LISTS" button in the tools tab.
4. Check the cache folder - the cache will be cleared.

**Expected behavior**

When the incompatible plugins list is the _only_ one updated we should not clear the whole cache.

The cache should still be cleared if any of the additional two lists are updated alongside the incompatible plugins list.

**Screenshots**

N/A

**Additional context**

**Slack discussion:** https://wp-media.slack.com/archives/CUT7FLHF1/p1690877516313079

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by engahmeds3ed**
## Description

As groomed.

Fixes #6077

## Type of change

*Please delete options that are not relevant.*

- Enhancement (non-breaking change which improves an existing functionality).

## Is the solution different from the one proposed during the grooming?

No, exactly the same.

# Checklists

## Generic development checklist

- [x] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [ ] I have added unit and integration tests that prove my fix is effective or that my feature works.
- [x] The CI passes locally with my changes (including unit tests, integration tests, linter).
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] If applicable, I have made corresponding changes to the documentation. *Provide a link to the documentation.*

## Test summary

- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I validated all Acceptance Criteria of the related issues. (If applicable, provide proof).
- [ ] I validated all test plan the QA Review asked me to.

### CODE REVIEW

**Comment by engahmeds3ed** [comment]
For the tests, I invested some time trying to add them quickly but I found that we now only test the default lists so we need to refactor those tests to be able to test the cache clear functionality with incompatible plugins only in a better way.

For that I created this issue not to forget about it: https://github.com/wp-media/wp-rocket/issues/6180

File: inc/Engine/Optimization/DynamicLists/DynamicLists.php [hunk 1]
```diff
@@ -149,7 +152,7 @@ public function update_lists_from_remote() {
 			 *
 			 * @since 3.12.1
 			 */
-			do_action( 'rocket_after_save_dynamic_lists' );
+			do_action( 'rocket_after_save_dynamic_lists', $should_purge );
```
Comment (remyperona) [inline inc/Engine/Optimization/DynamicLists/DynamicLists.php:157]: Add the new param description to the docblock

**Review by remyperona — CHANGES_REQUESTED** [review]

File: inc/Engine/Optimization/DynamicLists/DynamicLists.php (hunk 1 above)
Comment (engahmeds3ed) [reply inc/Engine/Optimization/DynamicLists/DynamicLists.php:157]: Good catch, Many thanks.
Fixed

**Review by remyperona — APPROVED** [review]

**Review by vmanthos — APPROVED** [review]
Working as expected! 🙏 :tophat:

**Pull request merged**